### PR TITLE
[Serve] Fix flaky standalone 2 test

### PR DESCRIPTION
**Pull request by GeneDer**
## Why are these changes needed?

`test_standalone_2.py` tests failed starting with `test_controller_recover_and_delete` not waiting for all deployment to shutdown and failed to clean up Ray and Serve for the rest of the tests. `test_controller_recover_and_delete` should be addressed by https://github.com/ray-project/ray/pull/37344. This PR cleans up Ray and Serve between each tests to improve robustness. 

## Related issue number

Fixes  flaky standalone 2 test cascading failures https://buildkite.com/ray-project/oss-ci-build-branch/builds/4933#01894a90-b11c-4aa5-a493-54c5bfdd3cc5
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
@GeneDer the diff is basically impossible to review because of the rename. Can you rebase and use `git mv` before making any edits so the diff renders properly? Or just revert the name change.

**Comment by edoakes** [comment]
Ignore my previous comment, I misread the diff

File: python/ray/serve/tests/test_deploy_app.py [hunk 1]
```diff
@@ -0,0 +1,1285 @@
...
+
+def _check_ray_stop():
+    try:
+        requests.get("http://localhost:52365/api/ray/version")
+        return False
+    except Exception:
+        return True
+
+
+class TestDeployApp:
```
Comment (edoakes) [inline python/ray/serve/tests/test_deploy_app.py:74]: this class can be removed now given that it's the only thing in the file

Comment (GeneDer) [inline python/ray/serve/tests/test_deploy_app.py:74]: @edoakes sorry I should have noted. This file is mostly just splitting this test class into it's own file.

File: python/ray/serve/tests/test_deploy_app.py [hunk 2]
```diff
@@ -0,0 +1,1285 @@
...
+    yield
+
+    subprocess.check_output(["ray", "stop", "--force"])
+    wait_for_condition(_check_ray_stop, timeout=15)
+
+
+@pytest.fixture(scope="class")
+def start_and_shutdown_ray_cli_class():
+    with start_and_shutdown_ray_cli():
+        yield
```
Comment (edoakes) [inline python/ray/serve/tests/test_deploy_app.py:63]: why do you need this level of nesting? can't you just put the code inside `start_and_shutdown_ray_cli` inside this fixture directly?

File: python/ray/serve/tests/test_deploy_app.py [hunk 3]
```diff
@@ -0,0 +1,1285 @@
...
+    try:
+        requests.get("http://localhost:52365/api/ray/version")
+        return False
+    except Exception:
+        return True
+
+
+class TestDeployApp:
+    @pytest.fixture(scope="function")
+    def client(self, start_and_shutdown_ray_cli_class, shutdown_ray_and_serve):
```
Comment (GeneDer) [inline python/ray/serve/tests/test_deploy_app.py:76]: This only change is in this client fixture how it clean between test

File: python/ray/serve/tests/test_deploy_app.py (hunk 2 above)
Comment (GeneDer) [reply python/ray/serve/tests/test_deploy_app.py:63]: I think the class scope helps to run this only once in the class. Else this test would need to further split out due to time out.

File: python/ray/serve/tests/test_deploy_app.py (hunk 1 above)
Comment (edoakes) [reply python/ray/serve/tests/test_deploy_app.py:74]: No I'm saying get rid of the class-level nesting since this whole file is just that class' tests

Comment (GeneDer) [reply python/ray/serve/tests/test_deploy_app.py:74]: oh I see, let me remove it

File: python/ray/serve/tests/test_deploy_app.py (hunk 3 above)
Comment (GeneDer) [reply python/ray/serve/tests/test_deploy_app.py]: This is no longer the case after dropping the class scope

**Review by edoakes — APPROVED** [review]

**Pull request merged**